draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1272,17 +1281,17 @@ a cryptographic hash function H which satisfies the following properties:
 1. The number of bits output by H should be b >= 2 * k for sufficient collision
 resistance, where k is the target security level in bits. (This is needed for a
 birthday bound of approximately 2^(-k).)
-2. H is modeled as a random oracle, so its output must be indistinguishable
-from a uniformly random bit string.
+2. H is modeled as a random oracle, so it should be chosen such that its output
+is plausibly indistinguishable from a uniformly random bit string.
```

Comment:
Great point. I've updated the wording.

BCIMRT10 don't discuss the underlying hash function, because in their construction they assume that one has already managed to instantiate a random oracle whose domain is the base field F.

For our purposes, length extension isn't a problem because SHA-2 gets wrapped inside of HMAC inside of the HKDF construction, which plausibly eliminates this worry. More generally, as long as we're happy modeling HKDF as a random oracle outputting L-byte strings for sufficiently large L (i.e., log p + k bits, k the security parameter), then hash_to_base is a random oracle whose domain is F giving at most 2^-k statistical distance from a uniform distribution over F.

Reply:
Agreed.